src/platform/silabs/BLEManagerImpl.h

```diff
@@ -157,7 +157,7 @@ class BLEManagerImpl final : public BLEManager, private BleLayer, private BlePla
     enum
     {
         kMaxConnections      = BLE_LAYER_NUM_BLE_ENDPOINTS,
-        kMaxDeviceNameLength = 16,
+        kMaxDeviceNameLength = 27,
```

Comment:
The device name goes into the scan response data which is 31 bytes. 
The response data array is 31 bytes and without the name contains 9 bytes already. With 27 bytes you are busting the array size and doing out-of-bound writes when doing the memcpy to add the name in the response data.

Reply:
Updated the value to 21 @jmartinez-silabs